src/raft.rs

```diff
@@ -1222,102 +1256,20 @@ impl<T: Storage> Raft<T> {
         Ok(())
     }
 
-    /// Apply a `BeginMembershipChange` variant `ConfChange`.
-    ///
-    /// > **Note:** This is an experimental feature.
-    ///
-    /// When a Raft node applies this variant of a configuration change it will adopt a joint
-    /// configuration state until the membership change is finalized.
-    ///
-    /// During this time the `Raft` will have two, possibly overlapping, cooperating quorums for
-    /// both elections and log replication.
-    ///
-    /// # Errors
-    ///
-    /// * `ConfChange.change_type` is not `BeginMembershipChange`
-    /// * `ConfChange.configuration` does not exist.
-    /// * `ConfChange.start_index` does not exist. It **must** equal the index of the
-    ///   corresponding entry.
     #[inline(always)]
-    pub fn begin_membership_change(&mut self, conf_change: &ConfChange) -> Result<()> {
-        if conf_change.get_change_type() != ConfChangeType::BeginMembershipChange {
-            return Err(Error::ViolatesContract(format!(
-                "{:?} != BeginMembershipChange",
-                conf_change.get_change_type()
-            )));
-        }
-        let configuration = if conf_change.has_configuration() {
-            conf_change.get_configuration().clone()
-        } else {
-            return Err(Error::ViolatesContract(
-                "!ConfChange::has_configuration()".into(),
-            ));
-        };
-        if conf_change.get_start_index() == 0 {
-            return Err(Error::ViolatesContract(
-                "!ConfChange::has_start_index()".into(),
-            ));
-        };
-
-        self.set_pending_membership_change(conf_change.clone());
-        let max_inflights = self.max_inflight;
-        self.mut_prs()
-            .begin_membership_change(configuration, Progress::new(1, max_inflights))?;
+    fn begin_membership_change(&mut self, cs: &ConfState) -> Result<()> {
```

Comment:
It's OK to rewrite/correct the documentation on these functions. They're non-trivial and could use explanation for our contributors.